### PR TITLE
fix: return enum type

### PR DESCRIPTION
**Pull request by aaqilniz**
`lb4 discover` doesn't handle enum properly. This PR implements [this](https://loopback.io/doc/en/lb4/Model.html#enum-property) workaround for the enum in `lb4 discover`. For that to work, MySQL connector needs to return the `enum` type as it is.

Related GitHub issues & PRs:

PR https://github.com/loopbackio/loopback-next/pull/10065

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [x] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

### CODE REVIEW

**Comment by aaqilniz** [comment]
Hi, @dhmlau. Please have a look at the PR. I am not sure what's causing the issue but the tests are failing for this PR too like [this](https://github.com/loopbackio/loopback-connector-mysql/pull/520) one.

**Comment by samarpanB** [comment]
CI check should pass after merging #539

**Comment by samarpanB** [comment]
@aaqilniz can you please rebase this PR ?

**Comment by aaqilniz** [comment]
Hi, @samarpanB. The tests are still failing.

**Comment by samarpanB** [comment]
@aaqilniz I did the same changes in a new PR #541 and it seemed to be clearing the checks. Can you try by creating a fresh new PR ? Or since you are now part of maintainers you can raise a PR in a new branch here itself without forking the repo. Try that. It seems something is wrong with github action on this PR or on this fork.

**Comment by samarpanB** [comment]
@aaqilniz rebase once more please, so that we can check if the CI action passes now.

**Comment by aaqilniz** [comment]
Hi, @dhmlau. The tests are finally passing for this PR too. Can you please have a look at the PR please?

**Review by achrinza — APPROVED** [review]

**Pull request merged**
